src/TreeHouse/Queue/Amqp/Driver/Amqp/Publisher/AmqpMessagePublisher.php

```diff
@@ -59,7 +59,11 @@ public function publish(Message $message, \DateTime $date = null, $flags = Excha
         $body = $message->getBody();
         $route = $message->getRoutingKey();
         $props = $message->getProperties()->toArray();
-
+        
+        if (!$this->exchange->getConnection()->isConnected()) {
```

Comment:
I don't think this is the right place for this. The publisher just needs to publish. This fix is useful in a specific context (long-running processes), I think you should incorporate it into your process instead.